src/main/java/org/opensearch/neuralsearch/mappingtransformer/SemanticInfoConfigBuilder.java

```
@@ -108,17 +109,91 @@ public Map<String, Object> build() {
         }
     }
 
+    private void validate() {
+        if (semanticFieldSearchAnalyzer != null && RankFeaturesFieldMapper.CONTENT_TYPE.equals(embeddingFieldType) == false) {
+            throw new IllegalArgumentException(
+                String.format(
+                    Locale.ROOT,
+                    "Cannot build the semantic info config because the embedding field type %s cannot build with semantic field search analyzer %s",
+                    embeddingFieldType,
+                    semanticFieldSearchAnalyzer
+                )
+            );
+        }
+
+        if (denseEmbeddingConfig != null && RankFeaturesFieldMapper.CONTENT_TYPE.equals(embeddingFieldType)) {
+            throw new IllegalArgumentException(
+                String.format(
+                    Locale.ROOT,
+                    "Cannot build the semantic info config because %s is not supported by %s.",
+                    DENSE_EMBEDDING_CONFIG,
+                    embeddingFieldType
+                )
+            );
+        }
+    }
+
     private Map<String, Object> buildKnnFieldConfig() {
         final Map<String, Object> config = new HashMap<>();
         config.put(TYPE, KNNVectorFieldMapper.CONTENT_TYPE);
         config.put(KNN_VECTOR_DIMENSION_FIELD_NAME, this.embeddingDimension);
+
         final Map<String, Object> methodConfig = new HashMap<>();
-        methodConfig.put(KNN_VECTOR_METHOD_NAME_FIELD_NAME, this.knnMethodName);
+
+        if (denseEmbeddingConfig != null) {
+            validateDenseEmbeddingConfig(denseEmbeddingConfig);
+            config.putAll(denseEmbeddingConfig);
+
+            final Object methodConfigObj = denseEmbeddingConfig.get(KNN_VECTOR_METHOD_FIELD_NAME);
```

should we throw an exception if `methodconfigObj` is null?